src/polyfills/intersection-observer.js

```diff
@@ -0,0 +1,239 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * See https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver.
+ */
+
+const UPGRADERS = '_upgraders';
+
+/**
+ * @param {!Window} win
+ */
+export function install(win) {
+  if (!win.IntersectionObserver) {
+    win.IntersectionObserver = /** @type {typeof IntersectionObserver} */ (IntersectionObserverStub);
+  }
+}
+
+/**
+ * @param {!Window} parentWin
+ * @param {!Window} childWin
+ */
+export function installForChildWin(parentWin, childWin) {
+  if (!childWin.IntersectionObserver && parentWin.IntersectionObserver) {
+    Object.defineProperties(childWin, {
+      IntersectionObserver: {get: () => parentWin.IntersectionObserver},
+      IntersectionObserverEntry: {
+        get: () => parentWin.IntersectionObserverEntry,
+      },
+    });
+  }
+}
+
+/**
+ * @param {!Window} win
+ * @return {boolean}
+ */
+export function shouldLoadPolyfill(win) {
+  return (
+    !win.IntersectionObserver ||
+    win.IntersectionObserver === IntersectionObserverStub ||
+    !win.IntersectionObserverEntry ||
+    // Some browsers implement `intersectionRatio`, but not `isIntersecting`.
+    // Polyfill smooths this out.
```

Comment:
If this is an important set of browsers, could we install a minimal polyfill here? It's essentially:

```js
Object.defineProperty(IntersectionObserverEntry.prototype, 'isIntersecting', {
  enumerable: true,
  configurable: true,
  get() {
    return this.intersectionRatio > 0;
  },
})

Reply:
Good idea. Done.